internal/handlers/pg/msg_explain.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pg
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/jackc/pgx/v4"
+
+	"github.com/FerretDB/FerretDB/internal/clientconn/conninfo"
+	"github.com/FerretDB/FerretDB/internal/handlers/common"
+	"github.com/FerretDB/FerretDB/internal/handlers/pg/pgdb"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+	"github.com/FerretDB/FerretDB/internal/util/version"
+	"github.com/FerretDB/FerretDB/internal/wire"
+)
+
+// MsgExplain implements HandlerInterface.
+func (h *Handler) MsgExplain(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
+	document, err := msg.Document()
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+	sp, err := h.parseExplainParams(ctx, document)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	var resDocs []*types.Document
+	err = h.pgPool.InTransaction(ctx, func(tx pgx.Tx) error {
+		var err error
+		resDocs, err = h.pgPool.Explain(ctx, tx, sp)
+		return err
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return h.buildExplainResult(ctx, document, resDocs)
+}
+
+// parseExplainParams validates document and returns pgdb.SQLParam.
+func (h *Handler) parseExplainParams(ctx context.Context, document *types.Document) (pgdb.SQLParam, error) {
```

Comment:
Could you please add some new lines to this method? It's hard to read it as is.
It's easier to read code when variables declaration or `switch` or `if` blocks are separated with a new line.

Reply:
> Could you please add some new lines to this method? It's hard to read it as is.

I've added, please see.